lib/bin/run_test.js

```diff
@@ -51,9 +51,13 @@ function invokeGeneratorOrPromiseFunction(fun) {
   }
 
   if (result && result.next) {  // This looks like a generator
-    return co(result);
+    return when.promise(function(resolve) {
+      // fixme: the promise returned by co seems to be
+      //        incompatible with the ones that when uses.
+      co(result).then(resolve);
+    });
   } else {
-    return when(result);
+    return when.resolve(result);
```

Comment:
Is this change there for a particular reason? I'd like to understand, that's all